src/lib/libast/path/pathbin.c

```diff
@@ -19,28 +19,30 @@
  *                     Phong Vo <phongvo@gmail.com>                     *
  *                                                                      *
  ***********************************************************************/
-/*
- * Glenn Fowler
- * AT&T Bell Laboratories
- *
- * return current PATH
- */
+//
+//  Glenn Fowler
+//  AT&T Bell Laboratories
+//
 #include "config_ast.h"  // IWYU pragma: keep
 
 #include <string.h>
 
 #include "ast.h"
 
+//  Return current PATH
+//
 char *pathbin(void) {
-    char *bin;
+    char *path;
+    // Used to restore value of saved astconf path from earlier function call
+    static char *astconf_path;
 
-    static char *val;
-
-    if ((!(bin = getenv("PATH")) || !*bin) && !(bin = val)) {
-        if (!*(bin = astconf("PATH", NULL, NULL)) || !(bin = strdup(bin))) {
-            bin = "/bin:/usr/bin:/usr/local/bin";
+    path = getenv("PATH");
```

Comment:
Yes, please continue to move assignments inside `if()` statements that can be easily hoisted.